File: test/fixtures/sample.php
<?php
/**
 * PHP Sample File for Method Extraction Testing
 */

class Calculator {
    private $value;

    public function __construct($initialValue = 0) {
        $this->value = $initialValue;
    }

    public function add($x, $y) {
        return $x + $y;
    }

    private function subtract($x, $y) {
        return $x - $y;
    }

    public static function multiply($x, $y) {
        return $x * $y;
    }

    protected function divide($x, $y) {
        if ($y === 0) {
            throw new Exception("Division by zero");
        }
        return $x / $y;
    }
}

function standalone_function($param) {
    return $param * 2;
}

function process_array(array $items) {
    return array_map(function($item) {
        return $item * 2;
    }, $items);
}

?>

File: test/fixtures/sample.rb
# Ruby Sample File for Method Extraction Testing

class Calculator
  attr_accessor :value

  def initialize(initial_value = 0)
    @value = initial_value
  end

  def add(x, y)
    x + y
  end

  def subtract(x, y)
    x - y
  end

  def self.multiply(x, y)
    x * y
  end

  def divide(x, y)
    raise "Division by zero" if y.zero?
    x / y
  end

  def question_method?
    @value > 0
  end

  def exclamation_method!
    @value = 0
  end

  private

  def private_helper
    @value * 2
  end
end

def standalone_function(param)
  param * 2
end

def process_block(&block)
  block.call if block_given?
end
